hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java

```diff
@@ -65,259 +77,229 @@ public static void initClass() throws Exception {
-      // We can't use list or valid if snapshot directory exists because DB
-      // transaction might not be flushed by the time.
-      Assert.assertNotNull(snapshotInfo);
-    } finally {
-      shell.close();
-    }
   }
 
   @Test
   public void testCreateSnapshotDuplicateName() throws Exception {
-    String volume = "vol-" + RandomStringUtils.randomNumeric(5);
-    String bucket = "buc-" + RandomStringUtils.randomNumeric(5);
-    String key = "key-" + RandomStringUtils.randomNumeric(5);
     String snapshotName = "snap-" + RandomStringUtils.randomNumeric(5);
 
-    String testVolBucket = OM_KEY_PREFIX + volume + OM_KEY_PREFIX + bucket;
-    String testKey = testVolBucket + OM_KEY_PREFIX + key;
+    int res = ToolRunner.run(shell,
+        new String[]{"-createSnapshot", bucketPath, snapshotName});
+    // Asserts that create request succeeded
+    Assertions.assertEquals(0, res);
 
-    createVolBuckKey(testVolBucket, testKey);
+    res = ToolRunner.run(shell,
+        new String[]{"-createSnapshot", bucketPath, snapshotName});
+    // Asserts that create request fails since snapshot name provided twice
+    Assertions.assertEquals(1, res);
+  }
 
-    OzoneFsShell shell = new OzoneFsShell(clientConf);
-    try {
-      int res = ToolRunner.run(shell,
-              new String[]{"-createSnapshot", testVolBucket, snapshotName});
-      // Asserts that create request succeeded
-      assertEquals(0, res);
-
-      res = ToolRunner.run(shell,
-              new String[]{"-createSnapshot", testVolBucket, snapshotName});
-      // Asserts that create request fails since snapshot name provided twice
-      assertEquals(1, res);
-    } finally {
-      shell.close();
-    }
+  /**
+   * Create snapshot should succeed.
+   * 1st case: valid snapshot name
+   * 2nd case: snapshot name length is less than 64 chars
+   */
+  @ParameterizedTest
+  @ValueSource(strings = {"snap-1",
+      "snap75795657617173401188448010125899089001363595171500499231286"})
+  public void testCreateSnapshotSuccess(String snapshotName)
+      throws Exception {
+    int res = ToolRunner.run(shell,
+        new String[]{"-createSnapshot", bucketPath, snapshotName});
+    // Asserts that create request succeeded
+    Assertions.assertEquals(0, res);
+
+    SnapshotInfo snapshotInfo = ozoneManager
+        .getMetadataManager()
+        .getSnapshotInfoTable()
+        .get(SnapshotInfo.getTableKey(volume, bucket, snapshotName));
+
+    // Assert that snapshot exists in RocksDB.
+    // We can't use list or valid if snapshot directory exists because DB
+    // transaction might not be flushed by the time.
+    Assertions.assertNotNull(snapshotInfo);
   }
 
-  @Test
-  public void testCreateSnapshotInvalidName() throws Exception {
-    String volume = "vol-" + RandomStringUtils.randomNumeric(5);
-    String bucket = "buc-" + RandomStringUtils.randomNumeric(5);
-    String key = "key-" + RandomStringUtils.randomNumeric(5);
-    String snapshotName = "snapa?b";
+  /**
```

Comment:
How case 3 is getting test? You are not passing or setting `bucketPath` it to invalid bucket? I'll suggest something like following:

```suggestion
  private static Stream<Arguments> createSnapshotFailureScenarios() {
    return Stream.of(
        Arguments.of("1st case: snapshot name contains invalid char",
            validBucketPath,
            "snapa?b"),
        Arguments.of("2nd case: snapshot name consists only of numbers",
            validBucketPath,
            "1234"),
        Arguments.of("3rd case: bucket path is invalid",
            invalidBucketPath,
            "validSnapshotName12"),
        Arguments.of("4th case: snapshot name length is more than 64 chars",
            validBucketPath,
            "snap156808943643007724443266605711479126926050896107709081166294")
    );
  }

  @ParameterizedTest(name = "{0}")
  @MethodSource("createSnapshotFailureScenarios")
  public void testCreateSnapshotFailure(String description,
                                        String bucketPath,
                                        String snapshotName)
      throws Exception {

    int res = ToolRunner.run(shell,
        new String[]{"-createSnapshot", bucketPath, snapshotName});
    Assertions.assertEquals(1, res);
  }
```

Also is there a way to validate the response from OFS. Because we are just validating that request fails but not verify the reason. We should verify that to make sure that reason is correct.

Reply:
> Also is there a way to validate the response from OFS. Because we are just validating that request fails but not verify the reason. We should verify that to make sure that reason is correct.

We can use `GenericTestUtils.SystemOutCapturer`, I'll pass the expected message in the parameters and fix it.